### PR TITLE
Avoid failing test on cleanup when filesystem has issues

### PR DESCRIPTION
**Pull request by abellina**
Signed-off-by: Alessandro Bellina <abellina@nvidia.com>

Closes https://github.com/NVIDIA/spark-rapids/issues/6515.

This fixes a sporadic issue in CI where we have seen failures when cleaning up temporary shuffle data, when what looks like NFS is failing to delete. We create a new temporary directory for each test, and we are not writing a ton of data, so I feel is fairly safe to ignore this.

### CODE REVIEW

**Review by tgravescs — DISMISSED** [review]
I'm assuming this is tempDir so at the end of the tests this ends up being cleaned up anyway, right?

**Comment by abellina** [comment]
> I'm assuming this is tempDir so at the end of the tests this ends up being cleaned up anyway, right?

No this is created using Spark's `Utils.createDirectory`. It is not a java temp dir. It is normally cleaned by the `SparkContext` on shut down. Do we want to change this to be a java temp directory? Might be better...

The function I changed here was copied verbatim from Spark's `BypassMergeSortShuffleWriterSuite.scala`

**Comment by tgravescs** [comment]
I'd say as long as its cleaned when SparkContext shut down it should be fine, if we run into issue we can revisit.

File: tests/src/test/320+-noncdh-nondb/scala/org/apache/spark/sql/rapids/RapidsShuffleThreadedWriterSuite.scala [hunk 1]
```diff
@@ -239,10 +240,16 @@ class RapidsShuffleThreadedWriterSuite extends FunSuite
...
   override def afterEach(): Unit = {
     TaskContext.unset()
+    blockIdToFileMap.clear()
+    temporaryFilesCreated.clear()
     try {
       Utils.deleteRecursively(tempDir)
-      blockIdToFileMap.clear()
-      temporaryFilesCreated.clear()
+    } catch {
+      case t: Throwable =>
```
Comment (gerashegalov) [inline tests/src/test/320+-noncdh-nondb/scala/org/apache/spark/sql/rapids/RapidsShuffleThreadedWriterSuite.scala:248]: consider limiting to `IOException` or `NonFatal(t)`

**Review by pxLi — APPROVED** [review]

**Comment by abellina** [comment]
>  I'd say as long as its cleaned when SparkContext shut down it should be fine, if we run into issue we can revisit.

Verified that the directories would be deleted by the shutdown hook if I don't:

```
22/09/29 12:11:32.492 shutdown-hook-0 INFO ShutdownHookManager: Shutdown hook called
22/09/29 12:11:32.492 shutdown-hook-0 INFO ShutdownHookManager: Deleting directory spark-a15133fc-e34a-4120-b98a-8bf8e44bc19a
...
```

Also this hook catches all `Exception` and throws them away (but logs them).

File: tests/src/test/320+-noncdh-nondb/scala/org/apache/spark/sql/rapids/RapidsShuffleThreadedWriterSuite.scala (hunk 1 above)
Comment (abellina) [reply tests/src/test/320+-noncdh-nondb/scala/org/apache/spark/sql/rapids/RapidsShuffleThreadedWriterSuite.scala]: @gerashegalov handled here: https://github.com/NVIDIA/spark-rapids/pull/6640/commits/b654e6dfbdf97478c920eb358fcddc2c80cc7de2

**Comment by abellina** [comment]
build

**Comment by abellina** [comment]
> Could not transfer artifact org.apache.maven:maven-settings-builder:jar:3.0.3 from/to central (https://repo.maven.apache.org/maven2): transfer failed for https://repo.maven.apache.org/maven2/org/apache/maven/maven-settings-builder/3.0.3/maven-settings-builder-3.0.3.jar.

retriggering

**Comment by abellina** [comment]
build

**Comment by abellina** [comment]
@tgravescs @gerashegalov mind taking another look?

**Review by tgravescs — APPROVED** [review]

**Comment by abellina** [comment]
integration tests failed on:
```
329858 [2022-09-29T19:20:59.531Z] FAILED ../../src/main/python/explain_test.py::test_explain_bucketd_scan[ALLOW_NON_GPU(ANY)]

329842 [2022-09-29T19:20:59.531Z] converted = AnalysisException('java.lang.RuntimeException: The root scratch dir: /tmp/hive on HDFS should be writable. Current per...che.spark.sql.hive.HiveExternalCatalog.withClient(HiveExternalCatalog.s       cala:102)\n\t... 21 more\n', JavaObject id=o404)
```
I am retriggering.

**Comment by abellina** [comment]
build

**Review by gerashegalov — APPROVED** [review]
LGTM

**Pull request merged**